go.mod

```diff
@@ -3,8 +3,8 @@ module github.com/buildkite/agent/v3
 go 1.18
 
 require (
-	cloud.google.com/go v0.0.0-20170217213217-65216237311a
-	github.com/DataDog/datadog-go v3.7.2+incompatible
+	cloud.google.com/go v0.65.0
```

Comment:
I'm a little confused why all the dependencies changed in this PR. Could you add that to the PR description?

Reply:
this PR was extracted from the giga-PR that did all of this and also added opentelemetry, so I _think_ that this is a result of adding the opentel dependencies, and then they require updated other stuff? I could be wrong though. Regardless, i'll clean this up and make it so that we're only updating the Datadog stuff

Reply:
Cool, so I went back and made it so that A) We have minimal dependency changes, only the ones required by the dd-trace-go upgrade, and B) They all happen in the [`Update dd-trace-go`](https://github.com/buildkite/agent/pull/1631/commits/65f8d4fb1158a29dad3ce68387d43e021a88516e) commit.

I _think_ reason some of our direct dependencies (notably `cloud.google.com/go` and `github.com/DataDog/datadog-go`, but also some `golang.org/x/...` libraries) have changed is because both we and dd-trace-go depend on these libs, and dd-trace-go's dependent versions changed.

I've double checked all of the libs that have changed, and made sure that none of our use cases have been deprecated or removed.